libmscore/rangeannotation.cpp

```diff
@@ -0,0 +1,502 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2002-2011 Werner Schweer
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENCE.GPL
+//=============================================================================
+
+
+#include "rangeannotation.h"
+#include "textannotation.h"
+#include "measure.h"
+#include "sym.h"
+#include "chord.h"
+#include "note.h"
+#include "score.h"
+#include "staff.h"
+#include "part.h"
+#include "segment.h"
+#include "property.h"
+#include "element.h"
+#include "score.h"
+#include "stafftext.h"
+#include "system.h"
+#include "xml.h"
+#include "text.h"
+#include <QRectF>
+#include <QPainter>
+
+namespace Ms {
+
+//---------------------------------------------------------
+//   RangeAnnotation
+//---------------------------------------------------------
+
+RangeAnnotation::RangeAnnotation(Score* s)
+  : Spanner(s)
+      {
+      setFlags(ElementFlag::MOVABLE | ElementFlag::SELECTABLE | ElementFlag::HAS_TAG);
+      _score         = s;
+      }
+//---------------------------------------------------------
+//   RangeAnnotationSegment
+//---------------------------------------------------------
+
+RangeAnnotationSegment::RangeAnnotationSegment(Score* score)
+   : SpannerSegment(score)
+      {
+      setFlag(ElementFlag::ON_STAFF, true);
+      }
+
+//---------------------------------------------------------
+//   firstNoteRestSegmentX
+//    in System() coordinates
+//    returns the position just after the last non-chordrest segment
+//---------------------------------------------------------
+
+qreal RangeAnnotation::firstNoteRestSegmentX(System* system)
```

Comment:
Slurs use this function as well, and other spanners use something similar. It could indeed be moved to utils.cpp